lib/ansible/module_utils/compat/datetime.py

```diff
@@ -3,36 +3,52 @@
 
 from __future__ import annotations
 
-from ansible.module_utils.six import PY3
+import datetime as _datetime
+import typing as t
 
-import datetime
+from ansible.module_utils.common.warnings import deprecate
 
 
-if PY3:
-    UTC = datetime.timezone.utc
-else:
-    _ZERO = datetime.timedelta(0)
+_UTC = _datetime.timezone.utc
 
-    class _UTC(datetime.tzinfo):
-        __slots__ = ()
 
-        def utcoffset(self, dt):
-            return _ZERO
+def _utcfromtimestamp(timestamp: float) -> _datetime.datetime:
+    """Construct an aware UTC datetime from a POSIX timestamp."""
+    return _datetime.datetime.fromtimestamp(timestamp, _UTC)
 
-        def dst(self, dt):
-            return _ZERO
 
-        def tzname(self, dt):
-            return "UTC"
+def _utcnow() -> _datetime.datetime:
+    """Construct an aware UTC datetime from time.time()."""
+    return _datetime.datetime.now(_UTC)
 
-    UTC = _UTC()
 
+_deprecated_shims_map: dict[str, t.Callable[..., object] | _datetime.timezone] = {
+    'UTC': _UTC,
+    'utcfromtimestamp': _utcfromtimestamp,
+    'utcnow': _utcnow,
+}
 
-def utcfromtimestamp(timestamp):  # type: (float) -> datetime.datetime
-    """Construct an aware UTC datetime from a POSIX timestamp."""
-    return datetime.datetime.fromtimestamp(timestamp, UTC)
+# __all__ = ('UTC', 'utcfromtimestamp', 'utcnow')  # pylint: disable=undefined-all-variable
```

Comment:
Is there a reason to leave this comment here?

Reply:
I might've missed it. Thanks!

Reply:
Fixed. Marking as resolved.